release/nightly_tests/dataset/map_batches_benchmark.py

```diff
@@ -0,0 +1,128 @@
+from typing import Literal, Optional, Union
+
+import ray
+from ray.data._internal.compute import ActorPoolStrategy, ComputeStrategy
+from ray.data.dataset import Dataset
+
+from benchmark import Benchmark
+
+
+def map_batches(
+    input_ds: Dataset,
+    batch_size: int,
+    batch_format: Literal["default", "pandas", "pyarrow", "numpy"],
+    compute: Optional[Union[str, ComputeStrategy]] = None,
+    num_calls: Optional[int] = 1,
+) -> Dataset:
+
+    ds = input_ds
+    for _ in range(num_calls):
+        ds = ds.map_batches(
+            lambda ds: ds,
+            batch_format=batch_format,
+            batch_size=batch_size,
+            compute=compute,
+        )
+    return ds
+
+
+def run_map_batches_benchmark(benchmark: Benchmark):
+    input_ds = ray.data.read_parquet(
+        "s3://shuffling-data-loader-benchmarks/data/input_data_0.parquet.snappy"
+    )
+    lazy_input_ds = input_ds.lazy()
+
+    batch_formats = ["pandas", "numpy"]
+    batch_sizes = [64, 128, 256, 512, 1024, 2048, 4096, None]
+    num_calls_list = [1, 2, 4, 8, 16]
+
+    # Test different batch_size of map_batches.
+    for batch_format in batch_formats:
+        for batch_size in batch_sizes:
+            # TODO(chengsu): Investigate why NumPy with batch_size less than 512,
+            # took forever to finish.
+            if batch_format == "numpy" and batch_size is not None and batch_size < 512:
+                continue
+
+            num_calls = 2
```

Comment:
I'm assuming this is 2 calls instead of 1 to eliminate some variance in the benchmarking, I'm wondering if something like 3 or 5 would be better for that purpose.